addon/mixins/ember-hooks/index.ts

```diff
@@ -0,0 +1,126 @@
+import Mixin from '@ember/object/mixin';
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import { depsAreEqual } from './util';
+
+type MemoizedStateEntry = [any, any[] | null];
+
+export interface IHookStore {
+  memoizedState: {
+    [key: number]: MemoizedStateEntry
+  }
+}
+
+export interface IEmberHooksComponent extends Component {
```

Comment:
Commenting now just for clarity even though it's closed. It's usually pretty common practice to export many of the interfaces you create, so that if someone ever had a need they are there.